src/main/java/dev/morling/onebrc/CalculateAverage_ericxiao.java

```diff
@@ -48,7 +58,7 @@ public ProcessFileMap(long readStart, long readEnd, boolean firstRead, boolean l
             this.firstRead = firstRead;
         }
 
-        private final HashMap<KeySlice, int[]> hashMap = new HashMap<>();
+        private final Station[] stations = new Station[MAP_SIZE];
```

Comment:
we should convert this to an array of integer and a separate array of (string, hash). for the array of (string, hash) we will make it a stack so that we only need a length of ~400. in Main we loop over the stack at read the hashed positions from the int[] array.